src/SerializerPlugin.php

```diff
@@ -3,51 +3,59 @@
 namespace Micro\Plugin\Serializer;
 
 use Micro\Component\DependencyInjection\Container;
-use Micro\Framework\Kernel\Plugin\AbstractPlugin;
-use Micro\Plugin\Serializer\Business\SerializerFacade;
-use Micro\Plugin\Serializer\Business\SerializerFactoryInterface;
-use Micro\Plugin\Serializer\Business\SerializerFactoryProvider;
-use Micro\Plugin\Serializer\Business\SerializerFactoryProviderInterface;
+use Micro\Framework\Kernel\KernelInterface;
+use Micro\Framework\Kernel\Plugin\DependencyProviderInterface;
+use Micro\Plugin\Serializer\Business\Provider\SerializerProvider;
+use Micro\Plugin\Serializer\Business\Provider\SerializerProviderInterface;
+use Micro\Plugin\Serializer\Business\Serializer\SerializerInterface;
+use Micro\Plugin\Serializer\Facade\SerializerFacade;
+use Micro\Plugin\Serializer\Facade\SerializerFacadeInterface;
+use Micro\Plugin\Serializer\Plugin\SerializerAdapterPluginInterface;
 
-class SerializerPlugin extends AbstractPlugin
+class SerializerPlugin implements DependencyProviderInterface, SerializerAdapterPluginInterface
 {
     /**
-     * {@inheritDoc}
+     * {@inheritdoc}
      */
     public function provideDependencies(Container $container): void
     {
-        $container->register(
-            SerializerFacadeInterface::class, function (Container $container) {
-                return $this->createSerializerFacade($container);
+        $container->register(SerializerFacadeInterface::class, function (
+            KernelInterface $kernel
+        ) {
+            return $this->createSerializerFacade($kernel);
+        });
+    }
+
+    public function getPriority(): int
+    {
+        return -1;
+    }
+
+    public function createSerializer(): SerializerInterface
+    {
+        return new class implements SerializerInterface
+        {
```

Comment:
The class should be move to a some separate plugin.
For example, if I don't want to use the default serializer in the app at all..